driver/pace/driver/run.py

```diff
@@ -475,15 +468,25 @@ def total_time(self) -> timedelta:
             days=self.days, hours=self.hours, minutes=self.minutes, seconds=self.seconds
         )
 
+    @functools.cached_property
+    def do_dry_convective_adjustment(self) -> bool:
+        return self.dycore_config.do_dry_convective_adjustment
+
+    @functools.cached_property
+    def apply_tendencies(self) -> bool:
+        return self.do_dry_convective_adjustment or not self.dycore_only
+
     @classmethod
     def from_dict(cls, kwargs: Dict[str, Any]) -> "DriverConfig":
         initialization_type = kwargs["initialization_type"]
         if initialization_type == "serialbox":
-            initialization_class: Type[InitializationConfig] = SerialboxConfig
+            initialization_class = SerialboxConfig  # type: ignore
+        elif initialization_type == "predefined":
+            initialization_class = PredefinedStateConfig  # type: ignore
         elif initialization_type == "baroclinic":
-            initialization_class = BaroclinicConfig
+            initialization_class = BaroclinicConfig  # type: ignore
         elif initialization_type == "restart":
-            initialization_class = RestartConfig
+            initialization_class = RestartConfig  # type: ignore
```

Comment:
I'm curious why the type: ignore were needed here?

Reply:
I needed this locally. I'm not sure why, but my pre-commit keeps complaining about the type being wrong for initialization class. I tried changing pythong versions, reinstall etc, but it still does this, though I think might not for everyone

Reply:
I can tell it needs the type: ignore as the code is now, but it should work with the type hint on the first assignment that got removed. It was needed to indicate that this variable is the more generic base class type, and not always a SerialboxConfig. This is fine though.